catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -1037,6 +1076,38 @@ private SolrQuery operationToQuery(String operation, String propertyName, String
     return new SolrQuery(geoQuery);
   }
 
+  private SolrQuery operationToQuery(String operation, String wkt) {
+
+    if (!StringUtils.isNotEmpty(wkt)) {
+      throw new UnsupportedOperationException("Wkt should not be null or empty.");
+    }
+
+    String geoQuery =
+        "("
+            + resolver
+                .anyGeoFields()
+                .map(
+                    propertyName ->
+                        getMappedPropertyName(propertyName, AttributeFormat.GEOMETRY, false))
```

Comment:
❓ I believe this will return <property_name>_geo.  The field we actually want to query is <property_name>_geo_index right?  So for example with the location field, the actual spatial index for the location field is `location_geo_index` not `location_geo`.  The `location_geo` field stores the wkt but is not suitable for doing the actual spatial query.  I didn't see changes to the `getMappedPropertyName` method to account for this yet the `getField` method, which is used there, no longer accounts for mapping to the `_geo_index` version of the field.  Am I just missing how we end up mapping to this now?

Reply:
SchemaFieldResolved::getSpecialIndexSuffix adds the "_index" to geo fields.

Reply:
`SchemaFieldResolver` is only used by `SolrQueryFilterVisitor`, which is only used in the `PersistentStoreImpl`.  The main catalog query goes through this `SolrFilterDelegate` and the `DynamicSchemaResolver`.  However, after walking through it again, it does look like the getMappedPropertyName calls the DynamicSchemaResolver::getField which would also add the "_index" by calling it's version of getSpecialIndexSuffix.  Not sure why I missed the call to `getSpecialIndexSuffix` before. 🤷‍♂️